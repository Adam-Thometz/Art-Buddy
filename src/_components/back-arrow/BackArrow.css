.BackArrow {
  position: absolute;
  cursor: pointer;
  display: flex;
  flex-direction: column;
}

.BackArrow img {
  height: 65px;
  width: 73px;
}