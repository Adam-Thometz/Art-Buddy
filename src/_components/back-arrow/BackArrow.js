import React from "react";
import { useNavigate } from "react-router-dom";

import arrow from './arrow.png'

const BackArrow = () => {
  const navigate = useNavigate()
  const goBack = () => {
    navigate(-1);
  }
  return (
    <div className="MenuNavbar-back" onClick={goBack}>
        <img src={arrow} alt="" />
        <span>BACK</span>
      </div>
  );
};

export default BackArrow;