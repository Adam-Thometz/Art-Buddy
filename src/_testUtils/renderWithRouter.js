import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;