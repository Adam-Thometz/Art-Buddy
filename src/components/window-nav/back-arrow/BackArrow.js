import { useNavigate } from "react-router-dom";

import "./BackArrow.css";

import arrow from "./arrow.png";

const BackArrow = () => {
  const navigate = useNavigate();
  const goBack = () => navigate(-1);
  return (
    <button className="BackArrow" onClick={goBack}>
      <img className="BackArrow-img" src={arrow} alt="" />
      <span className="BackArrow-text">BACK</span>
    </button>
  );
};

export default BackArrow;
