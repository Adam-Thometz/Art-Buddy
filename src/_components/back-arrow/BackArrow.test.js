import React from "react";
import renderWithRouter from '../../_testUtils/renderWithRouter';
import BackArrow from './BackArrow'

describe('BackArrow component', () => {
  it('renders without crashing', () => {
    renderWithRouter(<BackArrow />);
  })

  it('matches the snapshot of the back arrow', () => {
    const { asFragment } = renderWithRouter(<BackArrow/>);
    expect(asFragment()).toMatchSnapshot();
  })
})