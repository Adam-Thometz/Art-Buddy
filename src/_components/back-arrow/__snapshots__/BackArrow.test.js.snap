// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackArrow component matches the snapshot of the back arrow 1`] = `
<DocumentFragment>
  <div
    class="MenuNavbar-back"
  >
    <img
      alt=""
      src="[object Object]"
    />
    <span>
      BACK
    </span>
  </div>
</DocumentFragment>
`;
